pkg/postgres/pgutils/scan.go

```diff
@@ -11,16 +11,12 @@ type Unmarshaler[T any] interface {
 }
 
 // ScanRows scan and Unmarshal postgres rows into object of type T.
+//
+// This function closes the rows automatically on return.
 func ScanRows[T any, PT Unmarshaler[T]](rows pgx.Rows) ([]*T, error) {
-	var results []*T
-	for rows.Next() {
-		msg, err := Unmarshal[T, PT](rows)
-		if err != nil {
-			return nil, err
-		}
-		results = append(results, msg)
-	}
-	return results, rows.Err()
+	return pgx.CollectRows(rows, func(r pgx.CollectableRow) (*T, error) {
```

Comment:
Did you confirm that pgx is calling `rows.Close`?  I suspect they are and they have a history of sometimes closing the rows and sometimes not.

Reply:
Yes, this is highlighted in godoc and I've checked the source https://github.com/jackc/pgx/blob/v5.7.2/rows.go#L425
I wonder why we didn't close rows here but in the caller

Reply:
> I wonder why we didn't close rows here but in the caller

That's a good question.  One lost to time.